src/rust/ide/src/model/project.rs

```diff
@@ -351,59 +338,72 @@ pub mod test {
 
     /// Sets up project controller using mock Language Server clients.
     /// Passed functions should be used to setup expectations upon the mock clients.
-    /// Additionally, an `event_stream` expectation will be setup for a binary protocol, as
-    /// project controller always calls it.
+    /// Additionally, an `event_stream` expectation will be setup for a binary protocol, and
+    /// `get_suggestion_database` for json protocol, as
+    /// project controller always calls them.
     pub fn setup_mock_project
     ( setup_mock_json   : impl FnOnce(&mut language_server::MockClient)
     , setup_mock_binary : impl FnOnce(&mut enso_protocol::binary::MockClient)
-    ) -> controller::Project {
+    ) -> impl Future<Output=Project> {
```

Comment:
if the method sets up all expectations needed to create a project model, then the returned future should always be ready to complete its execution (should never stall). Then we can manually poll and unwrap it and keep the old API.